packages/truffle-compile/index.js

```diff
@@ -98,8 +102,8 @@ var compile = function(sources, options, callback) {
     language: "Solidity",
     sources: {},
     settings: {
-      evmVersion: options.solc.evmVersion,
-      optimizer: options.solc.optimizer,
+      evmVersion: options.compilers.solc.settings.evmVersion,
+      optimizer: options.compilers.solc.settings.optimizer,
```

Comment:
Do we default `compilers.solc.settings` anywhere? Or are those guaranteed non-null because of the grandfathering above?

Regardless, maybe this should be more explicit, in case that grandfathering goes away.

Reply:
They're defaulted in the `truffle-config` [here](https://github.com/trufflesuite/truffle/pull/1083/files#diff-ea3ccf54753c9aaae2ba020420166f18R40).  I'm going to double-check that those are coming through but I think the integration tests would crash on the grandfathering line if that wasn't an object.

Reply:
Oh cool, I missed that.

Reply:
Yes it seems be working . . . this what the Supplier config looks like from a scenario test after it's been copied in the constructor:
```javascript
{
 "version": null,
 "versionsUrl": "https://solc-bin.ethereum.org/bin/list.json",
 "compilerUrlRoot": "https://solc-bin.ethereum.org/bin/",
 "dockerTagsUrl": "https://registry.hub.docker.com/v2/repositories/ethereum/solc/tags/",
 "cache": true,
 "settings": {
  "optimizer": {
   "enabled": false,
   "runs": 200
  },
  "evmVersion": "byzantium"
 }
}
```